gax/build.gradle

```diff
@@ -26,15 +26,11 @@ dependencies {
   shadowNoGuava libraries['maven.com_google_guava_guava']
 }
 
-ext.generatedOutputDir = file("${buildDir}/generated-resources")
-task generateProjectProperties {
-  ext.outputFile = file("${generatedOutputDir}/dependencies.properties")
-  outputs.file(outputFile)
-  doLast {
-    outputFile.text = "version.gax=${project.version}"
+processResources {
+  filesMatching('dependencies.properties') {
+    expand('project': project)
```

Comment:
I don't think this is doing the same thing? The original code is creating a new file called `dependencies.properties` with the following content:
```
version.gax=<current project version>
```

Reply:
I added `dependencies.properties` to resources. Take a look at that file. It's effectively the same thing and works for both Maven and Gradle.
It's confusing that the file is called `dependencies.properties` like the one in the root.

Reply:
Oh, gotcha. But I don't see `dependencies.properties` added in this PR?

Reply:
ah! That might explain why CI is failing.

Reply:
Hmm... should it be `expand('project.version': version)` (or `expand('project.version': project.version)` if you want to be more verbose). My understanding is that, expanding is a simple templating mechanism, and `project.version` is a key. (In Groovy, `method('key': value)` is basically `method(new Map() with one entry of key --> value`).

Reply:
I tried that, but Gradle doesn't like the period in `${project.version}` and attempts to find the `project` object. So, I had to do this. I'm not experienced with Gradle. So, let me know if there is a better way that works.

Reply:
Ah, I thought it was a simple textual substitution, but apparently, it's more elaborate templating where things are evaluated. This is working, so we are good.